### PR TITLE
[et] Add `linear-create-audit-issues` command

### PR DESCRIPTION
**Pull request by bbarthec**
# Command has been run and the results can be seen in [Linerar.app](https://linear.app/expo/project/audit-modules-51a1031475eb/ENG)

# ~This PR is only to demonstrate the behaviour of the script I've prepared. I want some feedback before I run it against our Linear.app workspace.~

# Why

For the sake of tracking the progress of auditing every Expo module I think it would be good to have some Linear.app-based structure of tasks.
I propose following:
```
Umbrella Issue
├ Audit expo-av
│   ├ Audit expo-av: first subtask
|   ¦
|   └ Audit expo-av: n-th subtask
├ Audit expo-barcode-scanner
|   ├ Audit expo-barcode-scanner: first subtask
|   ¦
¦   └ Audit expo-barcode-scanner: n-th subtask
└ Audit n-th Expo module
```

Once this command is executed against all available Expo module, I would manually add existing module's tasks to the proper umbrella issues.

# How

I've created another `expotools` command that is capable of creating issues with nested sub-issues.
The main configuration of issues is available in `tools/src/linear-create-audit-issues/Configuration.ts` file.

Screenshot 2021-12-28 at 15 56 41
Screenshot 2021-12-28 at 15 56 46
Screenshot 2021-12-28 at 15 57 09

# Test Plan

I was testing this command using separate Linear.app workspace.

# Side notes

- I had to replace `@expo/commander` with the newest `commander` package, because talking with Linear REST API was not working with the previous version. Some things are broken, because of this change, but as this PR does not have to land I don't see that as a problem 😉

### CODE REVIEW

File: tools/src/linear-create-audit-issues/Configuration.ts [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        {
+          title: (moduleName: string) => `Audit ${moduleName} - Android: rewrite to SweetAPI`,
+          description: `Convert this module Android code to be SweetAPI-based`,
+          labelNames: ['Module', 'Android'],
+        },
+        {
+          title: (moduleName: string) => `Audit ${moduleName} - iOS: rewrite to Swift`,
+          description: `Convert this module iOS code to Swift.`,
+          labelNames: ['Module', 'iOS'],
+        },
```
Comment (tsapeta) [inline tools/src/linear-create-audit-issues/Configuration.ts:153]: This task makes no sense — why would we rewrite to Swift and not to Sweet API? These two are actually the same.
```suggestion
```

File: tools/src/linear-create-audit-issues/Configuration.ts [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        {
+          title: (moduleName: string) => `Audit ${moduleName} - iOS: rewrite to SweetAPI`,
```
Comment (tsapeta) [inline tools/src/linear-create-audit-issues/Configuration.ts:155]:
```suggestion
          title: (moduleName: string) => `Audit ${moduleName} - iOS: rewrite to Sweet API`,
```

File: tools/src/linear-create-audit-issues/Configuration.ts [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+          description: `Ensure this module is properly implemented on the Web platform.`,
+          labelNames: ['Module', 'Web'],
+        },
+        {
+          title: (moduleName: string) => `Audit ${moduleName} - Android: rewrite to Kotlin`,
+          description: `Convert this module Android code to Kotlin.`,
+          labelNames: ['Module', 'Android'],
+        },
+        {
+          title: (moduleName: string) => `Audit ${moduleName} - Android: rewrite to SweetAPI`,
```
Comment (tsapeta) [inline tools/src/linear-create-audit-issues/Configuration.ts:145]:
```suggestion
          title: (moduleName: string) => `Audit ${moduleName} - Android: rewrite to Sweet API`,
```

File: tools/src/linear-create-audit-issues/Configuration.ts [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+          labelNames: ['Module', 'Android'],
+        },
+        {
+          title: (moduleName: string) => `Audit ${moduleName} - iOS: rewrite to Swift`,
+          description: `Convert this module iOS code to Swift.`,
+          labelNames: ['Module', 'iOS'],
+        },
+        {
+          title: (moduleName: string) => `Audit ${moduleName} - iOS: rewrite to SweetAPI`,
+          description: `Convert this module iOS code to be SweetAPI-based`,
```
Comment (tsapeta) [inline tools/src/linear-create-audit-issues/Configuration.ts:156]:
```suggestion
          description: `Convert this module iOS code to be Sweet API-based`,
```

File: tools/src/linear-create-audit-issues/Configuration.ts [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+          labelNames: ['Module', 'iOS'],
+        },
+        {
+          title: (moduleName: string) => `Audit ${moduleName} - JS tests`,
+          description: `Audit JS tests for this module. Check \`test-suite\` app and provide solid test coverage.`,
+          labelNames: ['Module', 'Testing'],
+        },
+        {
+          title: (moduleName: string) => `Audit ${moduleName} - Android tests`,
+          description: `Convert this module iOS code to be SweetAPI-based`,
```
Comment (tsapeta) [inline tools/src/linear-create-audit-issues/Configuration.ts:166]: Not related to the title 😉

File: tools/src/linear-create-audit-issues/Configuration.ts [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+          labelNames: ['Module', 'Android', 'Testing'],
+        },
+        {
+          title: (moduleName: string) => `Audit ${moduleName} - iOS tests`,
+          description: `Convert this module iOS code to be SweetAPI-based`,
```
Comment (tsapeta) [inline tools/src/linear-create-audit-issues/Configuration.ts:171]: Same here, not related to the title 😉

File: tools/src/linear-create-audit-issues/Configuration.ts [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+          labelNames: ['Module', 'iOS', 'Testing'],
+        },
+        {
+          title: (moduleName: string) => `Audit ${moduleName} - examples screen`,
+          description: `Audit example screen by checking all module's functionalities are presented.`,
+          labelNames: ['Module', 'QA', `Testing`],
+        },
+        {
+          title: (moduleName: string) => `Audit ${moduleName} - compare with similar libraries`,
+          description: `Compare this module with other libraries available in the RN ecosystem. Check the differences and possibly narrow the functionalities gap.`,
```
Comment (tsapeta) [inline tools/src/linear-create-audit-issues/Configuration.ts:181]: Also should check if there are any new possibilities offered by the new versions of iOS/Android.

**Review by tsapeta — COMMENTED** [review]
I checked it briefly and suggested some changes. 

> I had to replace @expo/commander with the newest commander package, because talking with Linear REST API was not working with the previous version

How come? 😄 They are completely unrelated to each other 😮

**Comment by bbarthec** [comment]
@tsapeta, thank you for going through this PR 👏 
Answering your question about `commander` package problem:
- `expotools` loads each command in a dynamic way
- for some reason loading in this way caused `@linear/sdk` package to TIMEOUT every call to their API 🤷 
- to be more precise: calling the API via `@linear/sdk` from the main `expotools` file worked fine, while calling it from the other files was failing every time 🤷 
- after changing/upgrading the `commander` it started to work fine 😁

**Comment by Kudo** [comment]
roughly go through the code and LGTM. the work is awesome 🔥

File: tools/src/GitHub.ts [hunk 8]
```diff
@@ -177,6 +178,57 @@ export async function getIssueAsync(issue_number: number) {
   return data;
 }
 
+/**
+ * Internally GitHub threats pull requests as issues, therefore this endpoints returns both open issues and open PRs.
```
Comment (brentvatne) [inline tools/src/GitHub.ts:182]:
```suggestion
 * Internally GitHub treats pull requests as issues, therefore this endpoints returns both open issues and open PRs.
```

File: tools/src/linear-create-audit-issues/Configuration.ts [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+            `Audit ${moduleName} - compare with similar libraries and explore new platforms' capabilities`,
+          description: `
+Compare this module with other libraries available in the RN ecosystem.
+Check the differences and possibly narrow the functionalities gap.
+Additionally, explore the possible new capabilities offered by each platform.
+`,
+          labelNames: ['Module'],
+        },
+        {
+          title: (moduleName: string) => `Audit ${moduleName} - bug bush GitHub issues`,
```
Comment (brentvatne) [inline tools/src/linear-create-audit-issues/Configuration.ts:185]:
```suggestion
          title: (moduleName: string) => `Audit ${moduleName} - bug bash GitHub issues`,
```

**Review by brentvatne — APPROVED** [review]
this is a neat idea to help us organize this effort! maybe before we run this we should do a pass over the last few hundred issues to make sure everything is labeled correctly, so maximize the quality of the data we're pulling in during the sync

**Review by barthap — APPROVED** [review]
Great job! 👏 I agree with Brent about labelling the most recent issues before running this script.

**Review by tsapeta — APPROVED** [review]

**Pull request closed without merging**